File: backend/app/api/v1/schemas/visualization.py
"""可视化接口请求模型。"""

from __future__ import annotations

from pydantic import BaseModel, Field


class PmlRequest(BaseModel):
    """生成 PML 请求。"""

    task_id: str = Field(..., description="任务 ID")
    affinity: float | None = Field(None, description="结合能标注（可选）")


class OpenPymolRequest(BaseModel):
    """打开本地 PyMOL 请求。"""

    task_id: str = Field(..., description="任务 ID")

File: frontend/src/api/visualization.js
import http from './http'

export function generatePml(taskId, affinity = null) {
  return http.post('/visualization/pml', { task_id: taskId, affinity })
}

export function openPymol(taskId) {
  return http.post('/visualization/pymol/open', { task_id: taskId })
}
